src/bootstrap/dist.rs

```diff
@@ -439,51 +440,52 @@ pub fn rust_src(build: &Build) {
         copy(&build.src.join(item), &plain_dst_src.join(item));
     }
 
-    // If we're building from git sources, we need to vendor a complete distribution.
-    if build.src_is_git {
-        // Get cargo-vendor installed, if it isn't already.
-        let mut has_cargo_vendor = false;
-        let mut cmd = Command::new(&build.cargo);
-        for line in output(cmd.arg("install").arg("--list")).lines() {
-            has_cargo_vendor |= line.starts_with("cargo-vendor ");
-        }
-        if !has_cargo_vendor {
+    // Create the version file
+    write_file(&plain_dst_src.join("version"), build.rust_version().as_bytes());
+
+    if let Subcommand::Dist { .. } = build.flags.cmd {
```

Comment:
Hm ok so if we want this use case to work then we may need to dig a little deeper. My intention was to *not* have a clause such as this because it means that a "source tarball" is now context-dependent if you're installing source or not. Ideally every single source tarball we produce is the same, regardless of the context.

IIRC, though, the rust-src component is a stripped down version of the actual source tarball, right? (i.e. it doesn't contain the vendor dir anyway?) If that's the case maybe we could separate these code paths so the `rust-src` component installation doesn't generate a vendor dir but the source tarball itself still unconditionally does?